src/main/java/org/opensearch/knn/index/query/BaseQueryFactory.java

```diff
@@ -42,6 +43,7 @@ public static class CreateQueryRequest {
         private float[] vector;
         private byte[] byteVector;
         private VectorDataType vectorDataType;
+        private Map<String, ?> methodParameters;
```

Comment:
Why use wild card instead of Object?

Reply:
The intent is to indicate that it contains multiple types and that it is already parsed. It's subtle and wouldn't have been my first choice but I still think its better than object. Since this will be object on compile time I don't see any overhead here unless I am missing something.

My initial though process was to store the type in `MethodParameter` enum  so we can cast it whenever we want but as of now there was no need since the responsibility for Faiss falls onto jni layer.